File: 29-BitwiseAND/Solution.cpp
#include <iostream>

using namespace std;


int main(){
    int t;
    cin >> t;
    for(int a0 = 0; a0 < t; a0++){
        int n;
        int k;
        cin >> n >> k;

        // I don't understand this perfectly...
        if ((k-1 | k) <= n) {
            cout << k-1 << endl;
        } else {
            cout << k-2 << endl;
        }
    }
    return 0;
}

File: 29-BitwiseAND/Solution.py
#!/bin/python3

import sys

t = int(input().strip())
for a0 in range(t):
    n,k = input().strip().split(' ')
    n,k = [int(n),int(k)]

    # I don't understand this perfectly...
    if ((k-1) | k) <= n:
        print(k-1)
    else:
        print(k-2)


